### PR TITLE
Enhancement for template response

### PR DESCRIPTION
**Pull request by janis91**
As discussed in #889 , I added the capability for passing required scripts and styles to the template response. The scripts and styles will be added to the template after that.

An example constructor call (scripts work the same):

```
$styles = array(
            'vendor' => array('select2/select2'),
            'app' => array('nextnotes' => array('font-awesome.min',
                'simplemde.min',
                'markdown.min',
                'style')
            )
        );
new TemplateResponse('nextnotes', 'main', array(), 'user', $styles);
```

I will provide the tests, if this is what you were expecting @rullzer.

### CODE REVIEW

**Comment by BernhardPosselt** [comment]
What would scripts look like?

**Comment by janis91** [comment]
full example:
````
$styles = array(
            'vendor' => array('select2/select2'),
            'app' => array('nextnotes' => array('font-awesome.min',
                'simplemde.min',
                'markdown.min',
                'style')
            )
        );
$scripts = array(
            'vendor' => array('select2/select2'),
            'app' => array('nextnotes' => array('notes',
                'tags',
                'view',
                'app')
            )
        );
new TemplateResponse('nextnotes', 'main', array(), 'user', $styles, $scripts);
````

**Comment by BernhardPosselt** [comment]
Hm I'd say the idea is good but that's one giant constructor. Think of inlining everything ;D

What about combining both arrays in another array, so instead of $scripts, $styles, $htmlImports, ... we would simply pass something like $assets where $assets is:

```php
[
    'scripts' => ...
    'styles' => ...
...
```

**Comment by rullzer** [comment]
Thanks for the PR @janis91 

seeing it now. I'm not a big fan of magic arrays to pass to constructors.
I'm mayb leaning more to a few functions on the template response:

`addVendorScript`, `addScript` etc.

That might make it more explicit and a bit more abstract?
Mmmm need to do some more thinking about that.

**Comment by janis91** [comment]
I could remove it from the constructor and add functions for every case. But recently I recognized the style and script functions in /lib/private/legacy/template/functions.php. Maybe we have to discuss if the approach proposed here is really a step forward?

**Comment by raghunayyar** [comment]
Hi, can you add labels to this PR to have a better clarity if it is a WIP or a complete PR?
Thanks.

**Comment by MorrisJobke** [comment]
@rullzer Anything new?

File: lib/public/AppFramework/Http/TemplateResponse.php [hunk 1]
```diff
@@ -61,14 +61,17 @@ class TemplateResponse extends Response {
...
 	/**
 	 * constructor of TemplateResponse
 	 * @param string $appName the name of the app to load the template from
 	 * @param string $templateName the name of the template
 	 * @param array $params an array of parameters which should be passed to the
 	 * template
 	 * @param string $renderAs how the page should be rendered, defaults to user
-	 * @since 6.0.0 - parameters $params and $renderAs were added in 7.0.0
+	 * @param array $styles an array of styles which sould be added to the template
+	 * @param array $scripts an array of scripts which should be added to the template
```
Comment (MorrisJobke) [inline lib/public/AppFramework/Http/TemplateResponse.php]: This then needs to be removed.

**Review by MorrisJobke — CHANGES_REQUESTED** [review]

**Comment by rullzer** [comment]
Mmmm I'm still not really sure here.
@janis91 maybe we should indeed for now stick with the stuff that is there. And revisit this when we get a decent template engine. Does that sound good?

**Comment by janis91** [comment]
Yeah I am with you!

**Comment by BernhardPosselt** [comment]
FYI I found this idea to be a quite interesting take on templates https://r.je/transphporm-php-templates.html

Not saying that we should adopt it but its interesting

**Comment by rullzer** [comment]
Ok lets close this then.

Thnaks for the link @BernhardPosselt

**Pull request closed without merging**